src/NLog/Targets/Wrappers/AsyncTargetWrapper.cs

```diff
@@ -222,7 +218,32 @@ protected virtual void StartLazyWriterTimer()
             {
                 if (this.lazyWriterTimer != null)
                 {
-                    this.lazyWriterTimer.Change(this.TimeToSleepBetweenBatches, Timeout.Infinite);
+                    if (this.TimeToSleepBetweenBatches <= 0)
+                    {
```

Comment:
Please add some internal logs, as this is something new and it would be easier to track if this change has some (unwanted) effects